### ISSUE TITLE
Opentimestamps info : OutOfMemoryError: Java Heap Space

### ISSUE THREAD

**Issue by yurikilian** [opentimestamps/java-opentimestamps#21]
Hi,

Detected an exception on running info method:
```
Caused by: java.lang.OutOfMemoryError: Java heap space
	at java.util.ArrayList.<init>(ArrayList.java:153)
	at org.bitcoinj.core.Transaction.parse(Transaction.java:556)
```
It occurs when the timestamp have more than 3 pending attestations.

Updated bitcoinj-core lib to 0.14.7 version and it solved because they no more use ArrayList at this point.

Can I update the lib? 
Do you have any clue about this issue? 

Thank you.

**Comment by RCasatta** [comment]
> Can I update the lib? 

PRs are welcome

> Do you have any clue about this issue?

No could you provide the .ots you are using to see if I can reproduce?

**Comment by cwitte4191** [comment]
I've also run into the issue with the info() method in the java client.  I also get OutOfMemory.  
IMO, the relevant part of the stacktrace was apparently truncated in the original post, so i'll include a full stacktrace below.  Note that strTree() recurses indefinitely, with parameters of indent:0, verbosity: false.

[ots_oomIssue.tar.gz](https://github.com/opentimestamps/java-opentimestamps/files/2742648/ots_oomIssue.tar.gz)

I've also attached a tar file with a json document that was used to request the verification (original.json), the original.ots file that was returned from the stamp request, and the upgrade.ots file that was upgraded successfully, but fails the info() call.

java.lang.OutOfMemoryError: Java heap space
                at java.util.ArrayList.<init>(ArrayList.java:152)
                at org.bitcoinj.core.Transaction.parse(Transaction.java:556)
                at org.bitcoinj.core.Message.<init>(Message.java:114)
                at org.bitcoinj.core.Message.<init>(Message.java:146)
                at org.bitcoinj.core.ChildMessage.<init>(ChildMessage.java:47)
                at org.bitcoinj.core.Transaction.<init>(Transaction.java:189)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:392)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:384)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.Timestamp.strTree(Timestamp.java:403)
                at com.eternitywall.ots.OpenTimestamps.info(OpenTimestamps.java:61)

**Comment by Udala** [comment]
I have also bumped into this bug. I attach a relevant ots file which produces the bug.
[simplicity.pdf.ots.zip](https://github.com/opentimestamps/java-opentimestamps/files/2824791/simplicity.pdf.ots.zip)
The problem occurs because bitcoinj version 0.13.1 does not check defensively if a random message (hash, prepended or appended hashes...) is a valid Bitcoin raw transaction (with no witness). It just gets the number of inputs (or outputs), which can be extremely high when decoded from that arbitrary message, so it crashes with OutOfMemoryError when creating the containing ArrayList.

**Solution**: Modify pom.xml at java-opentimestamps.
Change:
```
<dependency>
     <groupId>org.bitcoinj</groupId>
     <artifactId>bitcoinj-core</artifactId>
     <version>0.13.1</version>
     <scope>compile</scope>
</dependency>
```
for this:
```
<dependency>
     <groupId>org.bitcoinj</groupId>
     <artifactId>bitcoinj-core</artifactId>
     <version>0.14.7</version>
     <scope>compile</scope>
</dependency>
```
and rebuild project.

**Comment by yurikilian** [comment]
Thank you @Udala ! Great clue!

**Comment by RCasatta** [comment]
updated dependency version with 356c5a705204dd9d8ed190f59ade81958bcca796
and released maven lib 1.17
